onnxruntime/core/session/utils.cc

```diff
@@ -505,6 +506,38 @@ Status AddEpOptionsToSessionOptions(gsl::span<const OrtEpDevice* const> ep_devic
 
       ORT_RETURN_IF_ERROR(config_options.AddConfigEntry((prefix + ep_option_keys[j]).c_str(), ep_option_vals[j]));
     }
+
+    // add custom op domain provided by EP to the session options if any
+    OrtEpFactory* ep_factory = ep_device->ep_factory;
+    if (ep_factory) {
+      auto is_already_in_domains = [&](std::string& domain_name, std::vector<OrtCustomOpDomain*>& domains) {
+        for (auto ptr : domains) {
+          if (domain_name == ptr->domain_) {
+            return true;
+          }
+        }
+        return false;
+      };
+
+      if (ep_factory->CreateCustomOpDomains == nullptr) {
```

Comment:
`ep_factory->CreateCustomOpDomains` does not exist for ORT 1.23. Think we need a check here that skips a factory with version older than 24.

Reply:
Added the check.